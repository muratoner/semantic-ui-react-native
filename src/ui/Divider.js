import PropTypes from 'prop-types';
import React from 'react';
import { StyleSheet, Text, View, ViewPropTypes } from 'react-native';
import { DividerProps } from '../';
import Icon from '../icons/Icon';

const Component = (props: DividerProps) => {
	const { style, text, icon, color, ...attributes } = props;
	return (
		<View {...attributes} style={[styles.divider(color), style]}>
			<View style={styles.title}>
				{icon && <Icon style={styles.icon} {...icon} />}
				{text && <Text style={styles.text}>{text}</Text>}
			</View>
		</View>
	);
};

const styles = StyleSheet.create({
	divider: (borderColor) => ({
		borderBottomWidth: 1,
		borderColor: borderColor || 'rgba(0, 0, 0, .1)',
		width: '100%',
	}),
	text: {
		fontWeight: 'bold',
	},
	icon: {
		marginRight: 5,
	},
	title: {
		alignItems: 'center',
		flexDirection: 'row',
		alignSelf: 'center',
		backgroundColor: 'white',
		paddingHorizontal: 10,
		top: 8,
	},
});

Component.protoTypes = {
	...ViewPropTypes,
	text: PropTypes.string,
};

export default Component;
