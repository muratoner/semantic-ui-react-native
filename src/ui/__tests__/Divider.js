import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Divider from '../Divider';

describe('Divider Component', () => {
	it('should render without issues', () => {
		const component = shallow(<Divider />);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with color', () => {
		const component = shallow(<Divider color="#f9f9" />);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with text', () => {
		const component = shallow(<Divider text="Description" />);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with icon and text', () => {
		const component = shallow(<Divider text="Users" icon={{ name: 'user' }} />);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with style', () => {
		const component = shallow(
			<Divider text="Users" style={{ backgroundColor: 'red' }} />
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});
});
