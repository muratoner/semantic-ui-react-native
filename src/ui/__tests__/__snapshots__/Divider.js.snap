// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Component should render with color 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#f9f9",
        "marginBottom": 10,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "paddingHorizontal": 10,
        "top": 8,
      }
    }
  />
</View>
`;

exports[`Divider Component should render with icon and text 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 1,
        "borderColor": "rgba(0, 0, 0, .1)",
        "marginBottom": 10,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "paddingHorizontal": 10,
        "top": 8,
      }
    }
  >
    <Icon
      name="user"
      style={
        Object {
          "marginRight": 5,
        }
      }
    />
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Users
    </Text>
  </View>
</View>
`;

exports[`Divider Component should render with style 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 1,
        "borderColor": "rgba(0, 0, 0, .1)",
        "marginBottom": 10,
        "width": "100%",
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "paddingHorizontal": 10,
        "top": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Users
    </Text>
  </View>
</View>
`;

exports[`Divider Component should render with text 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 1,
        "borderColor": "rgba(0, 0, 0, .1)",
        "marginBottom": 10,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "paddingHorizontal": 10,
        "top": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Description
    </Text>
  </View>
</View>
`;

exports[`Divider Component should render without issues 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 1,
        "borderColor": "rgba(0, 0, 0, .1)",
        "marginBottom": 10,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "paddingHorizontal": 10,
        "top": 8,
      }
    }
  />
</View>
`;
